### PR TITLE
Adding CMEK

### PR DESCRIPTION
**Pull request by leticiawebb**
Adding customer managed encryption key documentation.

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit c7ff47ce269ac7cb88d495a81296bada35b7542f with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/1306/c7ff47ce269ac7cb88d495a81296bada35b7542f/ci_running.html)
The overall status of the commit is 🟢 success

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟢 success</td></tr>
<tr><td>Vercel</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>🟢 success</td></tr>
</table>

File: docs/en/cloud/manage/cmek.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+---
+sidebar_label: Customer Managed Encryption Keys
+slug: /en/cloud/manage/cmek
+title: Customer Managed Encryption Keys
+---
+
+# Customer Managed Encryption Keys (CMEK)
+
+ClickHouse Cloud enables customers to encrypt their services housed in AWS using their own AWS KMS key. We utilize AWS KMS keys to encrypt the virtual file system, then use a key you generate and manage to encrypt the AWS KMS key in a process known as envelope encryption. All the service needs for this to work is access your AWS KMS key to decrypt & encrypt the data encryption key.
```
Comment (DanRoscigno) [inline docs/en/cloud/manage/cmek.md:9]:
```suggestion
ClickHouse Cloud enables customers to encrypt their services housed in AWS using their own AWS KMS key. We utilize AWS KMS keys to encrypt the virtual file system, then use a key you generate and manage to encrypt the AWS KMS key in a process known as envelope encryption. All the service needs for this to work is access to your AWS KMS key to decrypt & encrypt the data encryption key.
```

File: docs/en/cloud/manage/cmek.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+9. Click Finish.
+10. Click the alias of the key you just created.
+11. Use the copy button to copy the ARN.
+
+## Option 2. Configure or Create a KMS key using a CloudFormation stack
+
+ClickHouse provides a simple Cloud Formation stack to deploy the AWS Policy for your key. This method supports both existing KMS keys and creation of new KMS keys for ClickHouse Cloud integration.
+
+**Use an existing KMS key**
```
Comment (DanRoscigno) [inline docs/en/cloud/manage/cmek.md:60]:
```suggestion
### Use an existing KMS key
```

File: docs/en/cloud/manage/cmek.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+1. Login to your AWS account.
+2. Visit [this link](https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/quickcreate?templateURL=https://s3.us-east-2.amazonaws.com/clickhouse-public-resources.clickhouse.cloud/cf-templates/cmek.yaml&stackName=ClickHouseBYOK&param_KMSCreate=false&param_ClickHouseRole=arn:aws:iam::576599896960:role/prod-kms-request-role) to prepare the CloudFormation template.
+3. Enter the ARNs of the KMS key(s) you want to use (comma separated with no spaces in between).
+4. Accept "I acknowledge that AWS CloudFormation might create IAM resources with custom names." and click __Create stack__.
+5. Make note of the `RoleArn` and the `KeyArn` in the stack output as you will need these for the next step.
+
+
+**Create a new KMS key**
```
Comment (DanRoscigno) [inline docs/en/cloud/manage/cmek.md:69]:
```suggestion
### Create a new KMS key
```

File: docs/en/cloud/manage/cmek.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+1. Login to your AWS account.
+2. Visit [this link](https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/quickcreate?templateURL=https://s3.us-east-2.amazonaws.com/clickhouse-public-resources.clickhouse.cloud/cf-templates/cmek.yaml&stackName=ClickHouseBYOK&param_KMSCreate=true&param_ClickHouseRole=arn:aws:iam::576599896960:role/prod-kms-request-role) to prepare the CloudFormation template.
+4. Accept "I acknowledge that AWS CloudFormation might create IAM resources with custom names." and click __Create stack__.
+5. Make note of the `KeyArn` in the stack output as you will need this for the next step.
+
+## Option 3. Create a KMS key via Terraform
+
+For advanced users who want to deploy the key via terraform, check out the AWS provider documentation [here](https://registry.terraform.io/providers/hashicorp/aws/3.26.0/docs/resources/kms_key).
```
Comment (DanRoscigno) [inline docs/en/cloud/manage/cmek.md:78]:
```suggestion
For users who want to deploy the key via Terraform, check out the AWS provider documentation [here](https://registry.terraform.io/providers/hashicorp/aws/3.26.0/docs/resources/kms_key).
```

Comment (DanRoscigno) [reply docs/en/cloud/manage/cmek.md:78]: also, That link points to an older version of the page, should we link to a newer version (does it have to be 3.26.0)?

File: docs/en/cloud/manage/cmek.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+# Step 2. Starting a Service with Customer Managed Encryption Keys
+
+1. Log into your ClickHouse Cloud account.
+2. Go to the Services screen if you are not already there.
+3. Click __New Service__.
+4. Select your Cloud provider, Region and name your service.
```
Comment (DanRoscigno) [inline docs/en/cloud/manage/cmek.md:85]: Will this work on Dev services, or do we  require Production?

**Review by DanRoscigno — CHANGES_REQUESTED** [review]

File: docs/en/cloud/manage/cmek.md (hunk 5 above)
Comment (leticiawebb) [reply docs/en/cloud/manage/cmek.md:85]: It is only available for production resources in AWS.

Comment (leticiawebb) [reply docs/en/cloud/manage/cmek.md:85]: It also appears to be sitting behind a feature flag. Confirming.

Comment (leticiawebb) [reply docs/en/cloud/manage/cmek.md:85]: Confirmed and updated the notice at the top of the doc.

File: docs/en/cloud/manage/cmek.md [hunk 6]
```diff
@@ -9,7 +9,7 @@ title: Customer Managed Encryption Keys
 ClickHouse Cloud enables customers to encrypt their services housed in AWS using their own AWS KMS key. We utilize AWS KMS keys to encrypt the virtual file system, then use a key you generate and manage to encrypt the AWS KMS key in a process known as envelope encryption. All the service needs for this to work is access to your AWS KMS key to decrypt & encrypt the data encryption key.
 
 :::note
-To enable this feature please contact [support](https://clickhouse.cloud/support). Customer managed encryption keys must be specified at the time the service is created. Existing services cannot use this option at this time.
+To enable this feature for production resources in AWS please contact [support](https://clickhouse.cloud/support). Customer managed encryption keys must be specified at the time the service is created. Existing services cannot use this option at this time.
```
Comment (DanRoscigno) [inline docs/en/cloud/manage/cmek.md:12]:
```suggestion
To enable this feature for ClickHouse Cloud Production services in AWS please contact [support](https://clickhouse.cloud/support). Customer managed encryption keys must be specified at the time the service is created. Existing services cannot use this option at this time.
```

**Review by DanRoscigno — APPROVED** [review]

**Pull request merged**